src/pages/ChangePassword.vue

```diff
@@ -29,7 +30,47 @@
 <script>
     import Config from '../config.js';
     import Store from '../store.js';
+
+    import mixin from '../mixins/mixins';
+    import constants from "../constants";
+
     export default {
-        name: 'changePassword'
+        name: 'changePassword',
+        mixins: [
+            mixin.messages
+        ],
+        data: function () {
+            return {
+                netMgr : Store.netMgr,
+                pwd : "",
+                pwdConfirm: "",
+                email: this.$route.query.email,
+                token: this.$route.query.token,
+            }
+        },
+        methods: {
+            onRequestChangePassword: function () {
+                return this.netMgr.apiPost('user/lost_password/reset', {
+                    'password' : this.pwd,
+                    'password_confirmation' : this.pwdConfirm,
+                    'email' : this.email,
+                    'token' : this.token
+                }, 
+                    function (response) {
+                        switch (response.status) {
+                            case 200 :
+                                this.$router.replace({ path : '/' });
+                                break;
+                            default   :
+                                // something other than a 200 (202?)
+                                this.setMessage(response.data.status.join(' '), constants.MESSAGE_STATUS);
```

Comment:
join defaults to "," if there's nothing, and  since our errors are sentences with fulls stops and it looks nicer to have a space after a fullstop as in "Blah Blah Blah Error One. Blah Blah Error 2."